src/utils.rs

```diff
@@ -85,37 +86,38 @@ pub fn big_endian_uint256_serialize<S>(x: &Uint256, s: S) -> Result<S::Ok, S::Er
 where
     S: Serializer,
 {
-    if x == &0u32.into() {
+    if x.is_zero() {
         s.serialize_bytes(&[])
     } else {
-        let bytes = x.to_bytes_be();
-        s.serialize_bytes(&bytes)
+        // this also drops leading zero bytes
+        s.serialize_bytes(&x.to_u8_array_be()[(x.lz() / 8)..])
```

Comment:
Could please explain this logic  "x.to_u8_array_be()[(x.lz() / 8)..]" ? Or just share a link to the explanation

Reply:
Tests fail if I don't do this, because there needs to be no leading zero bytes (e.g. we need 1234 not 00001234). The `x.lz() / 8` divides the number of leading zero bits by 8, so calculates the number of bytes to skip from the big endian array of bytes.